### PR TITLE
Implement clone

### PR DESCRIPTION
**Pull request by BitcoinZavior**
Resolves https://github.com/payjoin/rust-payjoin/issues/164

#[derive(Clone)] attribute for the following structs implemented:

Configuration
PrjUriRequest
Context
MaybeInputsOwned
MaybeMixedInputScripts
MaybeInputsSeen
OutputsUnknown
ProvisionalProposal
PayjoinProposal

Adding Clone to the structs will make the FFI code simpler, ref: https://github.com/LtbLightning/payjoin-ffi. Right now, most structs have just one function. When you call this function, it takes over the memory because it uses self and not &self. This works fine in Rust, but it's a problem for FFI where we need to use &self.

Without Clone, we can't call payjoin functions directly. We've been using mutexes and extra helper functions to get around this. If we implement Clone, we can copy structs when we need to, making the FFI code cleaner and easier to work with. This change will help avoid those extra workarounds thereby making the code easier to maintain and enhance.

### CODE REVIEW

**Comment by DanGould** [comment]
It looks like your formatter is different than CI here which uses [this](https://github.com/payjoin/rust-payjoin/blob/master/rustfmt.toml) and nightly `cargo fmt`

**Comment by BitcoinZavior** [comment]
> It looks like your formatter is different than CI here which uses [this](https://github.com/payjoin/rust-payjoin/blob/master/rustfmt.toml) and nightly `cargo fmt`

Got it! it should pass now.

File: payjoin/src/input_type.rs [hunk 1]
```diff
@@ -56,13 +56,17 @@ impl<'de> serde::Deserialize<'de> for InputType {
             let ty = match parts[0].strip_prefix("type=") {
                 Some("pubkey") => SegWitV0Type::Pubkey,
                 Some("script") => SegWitV0Type::Script,
-                _ => return Err(serde::de::Error::custom("invalid SegWitV0 type")),
+                _ => {
+                    return Err(serde::de::Error::custom("invalid SegWitV0 type"));
+                }
```
Comment (DanGould) [inline payjoin/src/input_type.rs:61]: like this one

**Review by DanGould — CHANGES_REQUESTED** [review]
I appreciate that the formatting is fixed now, but this PR still makes ~hundreds of lines of formatting changes that are unrelated to `#[derive(Clone)]`. May you limit this to semantic changes related to `Clone`?

File: payjoin/src/input_type.rs (hunk 1 above)
Comment (BitcoinZavior) [reply payjoin/src/input_type.rs:61]: Sure 👍

Comment (DanGould) [reply payjoin/src/input_type.rs:61]: fyi ohttp: 0.5.1 makes KeyConfig [Clone](https://docs.rs/ohttp/0.5.0/ohttp/struct.KeyConfig.html#impl-Clone-for-KeyConfig), so you might want to depend on that in your pr

**Pull request closed without merging**